core/conversion/conversion.cpp

```diff
@@ -173,18 +176,47 @@ void AddInputs(ConversionCtx* ctx, at::ArrayRef<const torch::jit::Value*> inputs
 
 void MarkOutputs(ConversionCtx* ctx, at::ArrayRef<const torch::jit::Value*> outputs) {
   for (auto out : outputs) {
-    std::string name = std::string("output_") + std::to_string(ctx->num_outputs);
     auto it = ctx->value_tensor_map.find(out);
-    // Leaves the potential for unused outputs to be populated with nullptr
-    // "safely"
-    TRTORCH_CHECK(
-        it != ctx->value_tensor_map.end() && it->second,
-        "No corresponding output TRT Tensor found for TorchScript output: " << out->debugName());
-    auto out_tensor = it->second;
-    out_tensor->setName(name.c_str());
-    ctx->net->markOutput(*out_tensor);
-    LOG_INFO(ctx->logger, "Marking Output " << out->debugName() << " named " << name << " in engine (ctx.MarkOutput)");
-    ctx->num_outputs += 1;
+    if (it == ctx->value_tensor_map.end()) {
+      if (ctx->evaluated_value_map.find(out) != ctx->evaluated_value_map.end()) {
+        auto out_ivalue = ctx->evaluated_value_map[out];
+        if (out_ivalue.isList()) {
```

Comment:
Yes. That's right. Initially I've been trying various split graphs which is when this was needed before our discussion. I've been testing it using trtorchexec and it worked with that. But the test cases use a different API to construct/run the graph and it would be a problem. Removed the code. Anyways the split graph testcases have unpack anyway. We shall look into better way of handling different output types later. (Related issue opened : [https://github.com/NVIDIA/TRTorch/issues/284](url))